### PR TITLE
[docs] add proxy_service.trust_x_forwarded_for option

### PR DESCRIPTION
**Pull request by greedy52**
Related PR

- #26623 
- <s>#28420</s>

### CODE REVIEW

File: docs/pages/access-controls/guides/ip-pinning.mdx [hunk 1]
```diff
@@ -15,8 +15,9 @@ This helps minimize the risk of compromised credentials being used from differen
 
 ## How it works
 <Admonition type="note">
-  **Observed IP** - IP of the client. Teleport records this from the direct connection of
-  the user or, if it's enabled, a load balancer through the PROXY protocol.
+  **Observed IP** - IP of the client. Teleport records this from the direct
+  connection of the user or, if it's enabled, a load balancer through the PROXY
```
Comment (zmb3) [inline docs/pages/access-controls/guides/ip-pinning.mdx:19]: I would reword this. It's not clear that "it" refers to proxy protocol as written.

Maybe: Teleport records this from the direct connection of the user or via PROXY protocol or "X-Forwarded-For' headers if these features are enabled at the load balancer.

File: docs/pages/includes/config-reference/proxy-service.yaml [hunk 2]
```diff
@@ -138,3 +138,12 @@ proxy_service:
...
       # viewport. Does not apply to session recording view.
       scrollback_lines: 1000
+
+    # Enables the Proxy service to take client source IPs from the
+    # "X-Forwarded-For" headers for web APIs.
+    #
+    # In order to prevent IP spoofing, it is important to ensure that the Proxy
+    # service is placed behind a layer 7 load balancer or reverse proxy that
+    # sets client IPs in the "X-Forwarded-For" headers. Requests with more than
+    # one IPs will also be rejected.
```
Comment (alexfornuto) [inline docs/pages/includes/config-reference/proxy-service.yaml:148]: I'm not sure what else is being referred to by "also".

File: docs/pages/reference/helm-reference/teleport-cluster.mdx [hunk 3]
```diff
@@ -535,6 +535,28 @@ else [`clusterName`](#clusterName) is used. Default port is 3024.
...
+| `bool` | `false`       | no        | `proxy_service.trust_x_forwarded_for` |
+
+`trustXForwardedFor` enables the Proxy service to take client source IPs from
+the "X-Forwarded-For" headers for web APIs.
+
+<Admonition type="warning">
+In order to prevent IP spoofing, it is important to ensure that the Proxy
+service is placed behind a layer 7 load balancer or reverse proxy that sets
+client IPs in the "X-Forwarded-For" headers. Requests with more than one IPs
+will also be rejected.
```
Comment (alexfornuto) [inline docs/pages/reference/helm-reference/teleport-cluster.mdx:551]: Same comment as above re "also".

Comment (greedy52) [reply docs/pages/reference/helm-reference/teleport-cluster.mdx]: helm chart doc is reverted as the related helm change https://github.com/gravitational/teleport/pull/28420 is cancelled. The new service config will be "embedded" in the new ingress change https://github.com/gravitational/teleport/commit/446bd1c05e318ecb68beb49b9f6cea2b9a4ccc47

**Review by alexfornuto — APPROVED** [review]

**Review by smallinsky — APPROVED** [review]
I have left one suggestion apart from that LGMT.
